gyp_uv.py

```diff
@@ -48,16 +48,17 @@ def run_gyp(args):
   # On msvs it will crash if it gets an absolute path.
   # On Mac/make it will crash if it doesn't get an absolute path.
   if sys.platform == 'win32':
-    args.append(os.path.join(uv_root, 'uv.gyp'))
-    common_fn  = os.path.join(uv_root, 'common.gypi')
-    options_fn = os.path.join(uv_root, 'options.gypi')
-    # we force vs 2010 over 2008 which would otherwise be the default for gyp
+    arg_path = uv_root
+    # we force vs 2010 over 2008 which would otherwise be the default for gyp,
+	# since vs2008 uses old format `.vcproj` files
     if not os.environ.get('GYP_MSVS_VERSION'):
       os.environ['GYP_MSVS_VERSION'] = '2010'
   else:
-    args.append(os.path.join(os.path.abspath(uv_root), 'uv.gyp'))
-    common_fn  = os.path.join(os.path.abspath(uv_root), 'common.gypi')
-    options_fn = os.path.join(os.path.abspath(uv_root), 'options.gypi')
+    arg_path = os.path.abspath(uv_root)
+
+  args.append(os.path.join(arg_path, 'uv.gyp'))
+  common_fn  = os.path.join(arg_path, 'common.gypi')
+  options_fn = os.path.join(arg_path, 'options.gypi')
```

Comment:
please put these changes in a separate commit, since they are  no longer required for MSVS 2017 support.